`teleterm.TestStart` flakiness

## Failure

#### Link(s) to logs

- https://console.cloud.google.com/cloud-build/builds;region=us-west1/24e2eeba-faec-4115-ac7f-e4f81b0fa36f?project=ci-account

#### Relevant snippet

```
FAIL: github.com/gravitational/teleport/lib/teleterm.TestStart
===================================================
OUTPUT github.com/gravitational/teleport/lib/teleterm.TestStart
===================================================
=== RUN   TestStart
time="2022-06-29T22:20:23Z" level=info msg="tsh daemon is listening on unix:///tmp/TestStart3333087946/001/teleterm.sock."
time="2022-06-29T22:20:23Z" level=info msg="Context closed, stopping service."
    teleterm_test.go:45: 
        	Error Trace:	teleterm_test.go:45
        	            				teleterm_test.go:48
        	Error:      	Received unexpected error:
        	            	shutting down due to API Server error
        	            		grpc: the server has been stopped
        	Test:       	TestStart
--- FAIL: TestStart (0.01s)
```

--- a/lib/teleterm/teleterm_test.go
+++ b/lib/teleterm/teleterm_test.go
@@ -18,6 +18,7 @@ import (
 	"context"
 	"fmt"
 	"testing"
+	"time"
 
 	"github.com/gravitational/teleport/lib/teleterm"
 
@@ -41,7 +42,11 @@ func TestStart(t *testing.T) {
 	}()
 
 	defer func() {
-		cancel() // Stop the server.
+		// Make sure Start() is called.
+		time.Sleep(time.Millisecond * 500)
+
+		// Stop the server.
+		cancel()
 		require.NoError(t, <-wait)
 	}()
 

